src/penalty_parameter_functions.jl

```diff
@@ -140,6 +137,10 @@ function is_variable_dependent(::Type{ProportionalPenaltyParameter})::Bool
     return true
 end
 
+function reset_penalty_parameter(r::ProportionalPenaltyParameter)::ProportionalPenaltyParameter
```

Comment:
This won't actually fix the problem.  This function needs to go through the internal data structures and delete in entries